chameleon/src/main/java/org/hibernate/omm/jdbc/MongoStatement.java

```diff
@@ -58,57 +42,53 @@ public MongoStatement(MongoDatabase mongoDatabase, ClientSession clientSession,
         this.connection = connection;
     }
 
-    private Document runCommand(Document command) {
-        return mongoDatabase.runCommand(clientSession, command);
-    }
-
     @Override
     public ResultSet executeQuery(String sql) throws SimulatedSQLException {
         Assertions.notNull("sql", sql);
         throwExceptionIfClosed();
-        Document command = Document.parse(sql);
-        Document commandResult = runCommand(command);
-        if (commandResult.getDouble(OK_FIELD) != 1.0) {
-            throw new CommandRunFailSQLException(commandResult);
-        }
-        return new MongoResultSet(commandResult);
+        BsonDocument command = BsonDocument.parse(sql);
+        MongoCollection<BsonDocument> collection = mongoDatabase.getCollection(command.getString("aggregate").getValue(),
+                BsonDocument.class);
+        List<BsonDocument> pipeline = command.getArray("pipeline").stream().map(BsonValue::asDocument).toList();
+        MongoCursor<BsonDocument> cursor = collection.aggregate(clientSession, pipeline).cursor();
+
+        return new MongoResultSet(cursor,
+                pipeline.get(pipeline.size() - 1).asDocument().getDocument("$project").keySet().stream().toList());
     }
 
     @Override
     public int executeUpdate(String sql) throws SimulatedSQLException {
         Assertions.notNull("sql", sql);
         throwExceptionIfClosed();
-        Document command = Document.parse(sql);
-        Document commandResult = runCommand(command);
-        if (commandResult.getDouble(OK_FIELD) != 1.0) {
-            throw new CommandRunFailSQLException(commandResult);
+        BsonDocument command = BsonDocument.parse(sql);
+
+        String commandName = command.getFirstKey();
+        MongoCollection<BsonDocument> collection = mongoDatabase.getCollection(command.getString(commandName).getValue(),
+                BsonDocument.class);
+        switch (commandName) {
+            case "insert":
+                BsonDocument document = command.getArray("documents").get(0).asDocument();
+                collection.insertOne(clientSession, document);
+                return 1;
+            case "update":
+                BsonDocument updateDocument = command.getArray("updates").get(0).asDocument();
+                // TODO: this should call updateMany or updateOne depending on the value of the multi field
+                UpdateResult updateResult = collection.updateMany(clientSession, updateDocument.getDocument("q"), updateDocument.getDocument("u"));
```

Comment:
I think the default should be updateOne not updateMany, but not entirely sure of HQL semantics here. Same for delete

Reply:
As per https://www.mongodb.com/docs/manual/reference/command/delete/, deletes denote a deletion statement collection, and within each statement, there is a 'limit' field to specify whether delete one or many.

Similarly, as per https://www.mongodb.com/docs/manual/reference/command/update/, updates denote an updation statement collection, and within each statement, there is a 'multi' field to specify whether update one or many.

I need more time to consider this. Yeah, we have retry benefits to tap into, but for the current initial stage it might be better to focus on code logic correctness.